src/Controller/CarController.php

```diff
@@ -204,37 +206,97 @@ public function new(Request $request, Security $security): Response
         ]);
-        
-        $form->handleRequest($request);
-        
-        if ($form->isSubmitted() && $form->isValid()) {
-            $response = $this->carService->updateCar($car);
-
-            if ($response->getStatusCode() === Response::HTTP_OK) {
-                return $this->redirectToRoute('app_car_index');
-            }
-            
-            return $this->render('car/edit.html.twig', [
-                'form' => $form->createView(),
-                'car' => $car,
-                'error' => $response->getContent(),
-            ]);
-        }
         return $this->render('car/edit.html.twig', [
+            'car' => $car,
             'form' => $form->createView(),
+        ]);
+    }
+    
+
+    #[Route('/cars/update/{id}', name: 'app_car_update', methods: ['PUT'])]
+public function update(#[ValueResolver(CarValueResolver::class)] Car $car, Request $request, ValidatorInterface $validator): Response
+{
+    $requestData = $request->request->all()['car_form'];
+
+     $constraints = [
+        'brand' => [
+            new Assert\NotBlank(['message' => 'Brand is required.']),
+            new Assert\Length(['max' => 255, 'maxMessage' => 'Brand cannot be longer than 255 characters.']),
+        ],
+        'model' => [
+            new Assert\NotBlank(['message' => 'Model is required.']),
+            new Assert\Length(['max' => 255, 'maxMessage' => 'Model cannot be longer than 255 characters.']),
+        ],
+        'year' => [
+            new Assert\NotBlank(['message' => 'Year is required.']),
+            new Assert\Range([
+                'min' => 1900,
+                'max' => 2100,
+                'notInRangeMessage' => 'Year must be between {{ min }} and {{ max }}.',
+            ]),
+        ],
+        'engineCapacity' => [
+            new Assert\NotBlank(['message' => 'Engine capacity is required.']),
+            new Assert\Positive(['message' => 'Engine capacity must be a positive number.']),
+        ],
+        'horsePower' => [
+            new Assert\NotBlank(['message' => 'Horse power is required.']),
+            new Assert\Positive(['message' => 'Horse power must be a positive number.']),
+        ],
+        'color' => [
+            new Assert\NotBlank(['message' => 'Color is required.']),
+        ],
+        'registrationDate' => [
+            new Assert\NotBlank(['message' => 'Registration date is required.']),
+            new Assert\Date(['message' => 'Invalid registration date format.']),
+        ],
+    ];
+
+    $errorMessages = [];
+
+    foreach ($constraints as $field => $fieldConstraints) {
+        $violations = $validator->validate($requestData[$field] ?? null, $fieldConstraints);
+        foreach ($violations as $violation) {
+            $errorMessages[] = $violation->getMessage();
+        }
+    }
+      if (!empty($errorMessages)) {
+        return $this->render('car/edit.html.twig', [
             'car' => $car,
+            'errors' => $errorMessages,
```

Comment:
addFlash and redirect to car edit (some error): 
            $this->addFlash('form_errors', $errors);

            // Redirect to the specific route
            return $this->redirectToRoute('form_page');